stdlib/src/math/polynomial.mojo

```diff
@@ -81,15 +81,10 @@ fn _horner_evaluate[
     Returns:
         The polynomial specified by the coefficients evaluated at value x.
     """
-    alias num_coefficients = len(coefficients)
-    alias c_last = coefficients[num_coefficients - 1]
-    alias c_second_from_last = coefficients[num_coefficients - 2]
-
-    var result = x.fma(c_last, c_second_from_last)
+    var result = x.fma(coefficients[-1], coefficients[-2])
```

Comment:
While this is equivalent in terms of the result it yields, because the indexing happens in a non-parameter context, this will potentially materialize the `coefficients` vector and indexing could happen at runtime.

We currently don't have a way to mark functions as "free of side effects" or "can be safely hoisted to parameter domain", so the compiler cannot assume that `List.__getitem__` would not, for example, print to stdout. So users have to explicitly mark compile-time computations as such, using `alias`.